external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -133,6 +137,10 @@ private[kafka010] class KafkaMicroBatchStream(
 
   override def toString(): String = s"KafkaV2[$kafkaOffsetReader]"
 
+  override def metrics(latestOffset: Optional[Offset]): ju.Map[String, String] = {
+    KafkaMicroBatchStream.metrics(latestOffset, latestPartitionOffsets)
```

Comment:
`latestOffset` matches to `latestConsumedOffset` of `KafkaMicroBatchStream.metrics`. So it looks like the latest consumed offset.

In the doc of `ReportsSourceMetrics`, when I read "Returns the metrics reported by the streaming source with respect to the latest offset", I thought the latest offset parameter is latest available offset.

Could you make it clear in the API doc that the passed in is latest consumed offset?

Reply:
Because `MicroBatchStream` has an API `latestOffset` which is latest offset available. I guess we need to be more clear about the API item to avoid confusing.

Reply:
Thanks for your suggestion! Yes, it's really really confusing. I've changed the doc and param name.